### DISCUSSION TITLE
DeprecationWarning: The `punycode` module is deprecated. Please use a userland alternative instead.

### DISCUSSION THREAD

**Discussion by mike-lloyd03** [renovatebot/renovate#26220] [Request Help] [answered]
### How are you running Renovate?

Self-hosted Renovate

### If you're self-hosting Renovate, tell us what version of Renovate you run.

37.89.7

### If you're self-hosting Renovate, select which platform you are using.

gitlab.com

### Was this something which used to work for you, and then stopped?

It used to work, and then stopped

### Describe the problem

I just started getting this error today after a system update. I'm not sure if it directly relates to Renovate since I'm not seeing anyone else report this issue.

Whenever I run any `renovate` command, I get the following warning:
```
(node:638127) [DEP0040] DeprecationWarning: The `punycode` module is deprecated. Please use a userland alternative instead.
(Use `node --trace-deprecation ...` to show where the warning was created)
```

I'm not sure how to resolve this issue as it doesn't look like Renovate uses punycode anywhere.

### Relevant debug logs

<details><summary>Logs</summary>

```

LOG_LEVEL=debug renovate --version
DEBUG: Using RE2 as regex engine
DEBUG: Parsing configs
DEBUG: Checking for config file in /home/renovate-bot/config.js
(node:639699) [DEP0040] DeprecationWarning: The `punycode` module is deprecated. Please use a userland alternative instead.
(Use `node --trace-deprecation ...` to show where the warning was created)
DEBUG: No config file found on disk - skipping
37.89.7

```

</details>

### Have you created a minimal reproduction repository?

Placeholder value, please select the correct response from the dropdown

**Comment by rarkins** [comment]
Multiple of Renovate dependencies have `punycode` as a transitive dependency. I'm guessing you're running using Node v21?

**Reply by mike-lloyd03** [reply]
Indeed I am.

**Reply by rarkins** [reply]
The message doesn't appear in Node 20, so use that instead if it annoys you. We have no way to remove it from all of our transitive dependencies

**Reply by mike-lloyd03** [reply]
Understood. Thanks for sorting this out for me.

**Reply by mike-lloyd03** [reply]
I just downgraded to Node 20 and all is well.

**Reply by Letsplaybar** [reply]
Is this still the current solution? Because how future-proof is that then, in the meantime 22 is lts and 20 runs for about 1.5 years until it is EOL, so the recommended solution can't be yes downgarde to 20, unfortunately we can't do anything else.

**Reply by rarkins** [reply]
Nobody said Node v20 forever

**Comment by viceice** [comment, marked as answer]
- https://github.com/renovatebot/renovate/issues/32395
